scripts/build_ses.sh

```diff
@@ -0,0 +1,309 @@
+#
+# The nodes are the crowbar machine names. Because SES uses deepsea, the first node provided
+# will become the salt master. So any other way of viewing the argument list of nodes are as:
+#    <salt master> <salt minion> [<salt minion> ..]
+#
+# This script is used in the function onadmin_external_ceph by qa_crowbarsetup.sh as apart of
+# the deployexternalceph step.
+#
+# For further ses5 install instructions on which this was based
+# see: http://beta.suse.com/private/SUSE-Storage-Beta/doc/ses-manual_en/ceph.install.saltstack.html
+NUM_MASTER=1
+NUM_ADMIN=1
+NUM_MON=3
+NUM_MDS=1
+NUM_IGW=0
+NUM_RGW=2
+
+ip_cidr_regex='^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$'
+
+if (( $# < 2 ))
+then
+    echo "Usage: $0 [<public_network/cidr> <cluster_network/cidr>] <salt master> <salt minion> [, <salt minion>, ..]"
+    exit 1
+fi
+
+if [[ "$1" =~ $ip_cidr_regex ]]
+then
+    public_network=$1
+    shift
+    if [[ "$1" =~ $ip_cidr_regex ]]
+    then
+        cluster_network=$1
+        shift
+    fi
+fi
+master=$1
+shift
+minions=($@)
+
+function run_ssh_cmd() {
+    host=$1
+    shift
+    ssh $host -C "$@"
+    if (( $? > 0 ))
+    then
+        echo "Ssh command '$@' failed on host '$host'"
+        exit 2
+    fi
+}
+
+function salt_setup() {
+    # install salt master
+    run_ssh_cmd $master "zypper -n in deepsea"
+    run_ssh_cmd $master "systemctl enable salt-master.service"
+    run_ssh_cmd $master "systemctl start salt-master.service"
+    run_ssh_cmd $master "echo 'master: $master' > /etc/salt/minion.d/master.conf"
+    run_ssh_cmd $master "sudo systemctl enable salt-minion.service"
+    run_ssh_cmd $master "sudo systemctl start salt-minion.service"
+    if ssh $master "[ ! -f /srv/pillar/ceph/deepsea_minions.sls.bak ]  && [ -f /srv/pillar/ceph/deepsea_minions.sls ]"
+    then
+        run_ssh_cmd $master "sudo cp /srv/pillar/ceph/deepsea_minions.sls /srv/pillar/ceph/deepsea_minions.sls.bak"
+        cat <<EOF | ssh $master "cat > /srv/pillar/ceph/deepsea_minions.sls"
+# Using the deepsea grain 'G@deepsea:*' wasn't working, so setting to all nodes.
+# original file saved to /srv/pillar/ceph/deepsea_minions.sls.bak
+deepsea_minions: '*'
+EOF
+    fi
+
+    # install salt minion on the minions
+    for minion in "${minions[@]}"
+    do
+        run_ssh_cmd $minion "sudo zypper -n in salt-minion"
+        run_ssh_cmd $minion "sudo systemctl enable salt-minion.service"
+        run_ssh_cmd $minion "echo 'master: $master' > /etc/salt/minion.d/master.conf"
+        run_ssh_cmd $minion "sudo systemctl start salt-minion.service"
+    done
+
+    # Back on master, accept all the minions salt keys
+    run_ssh_cmd $master "salt-key --accept-all -y"
+    echo -n "waiting on keys (wait 20 times in 2 sec increments)."
+    for x in $(seq 20)
+    do
+      if (( $(run_ssh_cmd $master "salt-key -l acc |grep -cv 'Accepted Keys:'") != $(( ${#minions[@]} + 1 )) ))
```

Comment:
bashate wants indention to be multiples of 4 here.

Reply:
Thanks,
And another in qa_crowbarsetup.sh in the new function. Must've done something stupid like hitting the backspace on a line I wasn't actually on or something. Anyway, this will teach me to do a `make test` even if it's only a minor update to the code.